examples/components/CCFRAUD/traininsilo/run.py

```diff
@@ -457,18 +548,17 @@ def main(cli_args=None):
     args = parser.parse_args(cli_args)
 
     print(f"Running script with arguments: {args}")
-    run(args)
+    print(f"CUDA available: {torch.cuda.is_available()}")
+    cuda_count = torch.cuda.device_count()
+    print(f"Found CUDA devices: {cuda_count}")
 
+    if torch.cuda.device_count() > 1:
+        print("Running DDP training to utilize multiple GPUs.")
+        mp.spawn(run_parallel, args=(cuda_count, args), nprocs=cuda_count, join=True)
```

Comment:
let's not use this to run DDP, but instead let's use the AzureML way of creating distributed jobs.

this would greatly simplify the code.

check https://github.com/Azure/azureml-examples/blob/main/tutorials/e2e-distributed-pytorch-image/e2e-object-classification-distributed-pytorch.ipynb
 section "Training a distributed gpu job"

Reply:
the implementation of the pytorch launcher is taken care of by AzureML, your scripts only has to run the DDP which will consume the env vars (MASTER_ADDR) set up for you by AzureML.

Reply:
I have checked out their solution and I'm not sure about it because either we make pipeline or the component messy. The reason for this is that if you have multi-gpu machine you still need to retrieve all details about it in the pipeline and I'm not really sure how to get the number of GPUs connected to that particular compute. If we make it as a parameter, it's going to be less user friendly.

Reply:
@jfomhover what do you think? Can we discuss this maybe in the next meeting?